web/src/pages/FilesPage.tsx

```diff
@@ -25,13 +66,26 @@ export const FilesPage: React.FC = () => {
 
   // Current directory info and details.
   const [entries, setEntries] = useState<Entry[]>();
-  const [selectedEntry, setSelectedEntry] = useState<Entry>();
   const [currentDir, setCurrentDir] = useState<Entry>();
+  const onClickEntry = useCallback(
+    async (target: string | Entry) => {
+      if (typeof target === 'string') {
+        history.push(`/files/${libraryId}/${encodeURIComponent(target)}`);
+      } else if (target.category === 'Folder') {
+        history.push(`/files/${libraryId}/${encodeURIComponent(resolvePath(target))}`);
```

Comment:
Currently when navigating into a folder you can briefly see the details panel switch back to showing the current directory info (more noticeable the slower the connection). Calling `setCurrentDir` here with the entry we're navigating to will keep it selected, while the `refresh` call retrieves the entry from the API.

```suggestion
        history.push(`/files/${libraryId}/${encodeURIComponent(resolvePath(target))}`);
        setCurrentDir(target);
```

Reply:
Hm yeah, I am planning a big refactor because the dataflow and dependencies in these hooks are... confusing. It should be one-directional. But not in this branch :p